### PR TITLE
actionRules - AlertsManagement

### PR DESCRIPTION
**Pull request by roramc**
adding new resource type actionRules in AlertsManagement RP

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/examples/ActionRules_Delete.json
Comment (nschonni) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/examples/ActionRules_Delete.json]: This wouldn't be valid JSON

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (nschonni) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: SupressionType -> SuppressionType

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (nschonni) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: reccurrence -> recurrence

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/examples/ActionRules_Delete.json
Comment (roramc) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/examples/ActionRules_Delete.json]: resolved in latest commit

Comment (roramc) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/examples/ActionRules_Delete.json]: resolved in latest commit

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (roramc) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: resolved in latest commit

**Comment by praries880** [comment]
@roramc The [name](https://github.com/Azure/azure-rest-api-specs/pull/4955/files#diff-5f2fc3c12561cd22b023ac3b6f46697b) contains "privatepreview".. if this is indeed a private preview that why not use the [private repo](https://github.com/Azure/azure-rest-api-specs-pr) to drive this?

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (praries880) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: is this expected? should the version be `2018-11-02-preview`?

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (praries880) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: typo...

```suggestion
        "description": "List all action rules of the subscription and given input filters",
```

**Review by praries880 — CHANGES_REQUESTED** [review]

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (roramc) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: fixed in latest commit. Also, this preview version is just for the new resource type. The old resource types continue with the old version. So, should the old resource type definitions be removed from this .json file?

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (roramc) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: fixed in latest commit.

**Comment by praries880** [comment]
@roramc waiting on ARM review before I can approve the PR.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (ravbhatnagar) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: string enums generally work better longer term. Specially when you want to expand states the property can take.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (ravbhatnagar) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: Not sure if this kind of PII is ok to have on the resource response. I will double check on my side.

**Review by ravbhatnagar — COMMENTED** [review]
Only reviewed actionRule APIs since that seemed the new type that was added. Next time, please remember to make the first commit which is a copy of existing version. The new changes should come in the next commit onwards. It will make reviewing easier.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (roramc) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: The same object gets returned even when we make a GET call to list all action rules for a given subscription. At that point user would want to know what resource group this action rule is stored in. So we added this in the properties explicitly. If  conveying it through Id is enough, we could remove this property as well. Not sure what is the convention.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (roramc) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: Ideally we would want to show who created/updated the given resource. Aren't other RPs doing the same? Do let us know if there is any concern with this.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json
Comment (roramc) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json]: Resolved in latest commit

**Comment by praries880** [comment]
@ravbhatnagar do the latest changess from @roramc  address all the issues you raised?

**Comment by ravbhatnagar** [comment]
awaiting confirmation from service team. We recommended service team check with privacy since it contains potentially PII information.

**Review by KrisBash — APPROVED** [review]
Thanks for the changes. Signing off

**Review by praries880 — APPROVED** [review]

**Pull request merged**
